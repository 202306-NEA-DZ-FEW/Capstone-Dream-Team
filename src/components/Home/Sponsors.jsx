export default function Sponsors() {
    return (
        <div>
            <div class='flex h-full my-20 py-20 items-center justify-center bg-[#BCE4EB]'>
                <div class='w-full rounded-lg bg-white px-8 py-4 shadow-md m-4'>
                    <div class='px-1 py-4'>
                        <h3 className='max-w-lg mb-6 font-sans text-2xl text-center font-bold leading-none tracking-tight text-[#192655] sm:text-3xl md:mx-auto'>
                            Our sponsors
                        </h3>
                    </div>
                    <ul class='grid grid-cols-4 gap-2 px-1'>
                        <li class='flex items-center flex-col'>
                            <img
                                src='https://randomuser.me/api/portraits/women/11.jpg'
                                alt=''
                                class='rounded-full w-16 h-16 object-cover'
                            />
                            <h5 class='font-semibold'>Alex</h5>
                        </li>
                        <li class='flex items-center flex-col'>
                            <img
                                src='https://randomuser.me/api/portraits/women/11.jpg'
                                alt=''
                                class='rounded-full w-16 h-16 object-cover'
                            />
                            <h5 class='font-semibold'>Sarah</h5>
                        </li>
                        <li class='flex items-center flex-col'>
                            <img
                                src='https://randomuser.me/api/portraits/women/11.jpg'
                                alt=''
                                class='rounded-full w-16 h-16 object-cover'
                            />
                            <h5 class='font-semibold'>Jericho</h5>
                        </li>
                        <li class='flex items-center flex-col'>
                            <img
                                src='https://randomuser.me/api/portraits/women/11.jpg'
                                alt=''
                                class='rounded-full w-16 h-16 object-cover'
                            />
                            <h5 class='font-semibold'>Dianna</h5>
                        </li>
                    </ul>
                </div>
            </div>
        </div>
    );
}
