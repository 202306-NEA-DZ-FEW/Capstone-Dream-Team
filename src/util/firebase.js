// Import the functions you need from the SDKs you need
import { initializeApp } from "firebase/app";
import { getFirestore } from "firebase/firestore";
import { getAuth } from "firebase/auth";
// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
const firebaseConfig = {
    apiKey: process.env.NEXT_PUBLIC_FIREBASE_API_KEY,
    authDomain: "capstone-project-944c9.firebaseapp.com",
    projectId: "capstone-project-944c9",
    storageBucket: "capstone-project-944c9.appspot.com",
    messagingSenderId: "455661891383",
    appId: "1:455661891383:web:65bbc812478ffda076b4b2",
};

// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const db = getFirestore(app);
export const auth = getAuth(app);
