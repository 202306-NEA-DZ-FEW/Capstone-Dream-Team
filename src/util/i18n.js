import i18n from "i18next";
import arTranslations from "public/locales/ar/common.json";
import enTranslations from "public/locales/en/common.json";
import frTranslations from "public/locales/fr/common.json";
import { initReactI18next } from "react-i18next";

i18n.use(initReactI18next).init({
    resources: {
        en: enTranslations,

        ar: arTranslations,

        fr: frTranslations,
    },
    lng: "en", // Set the default language
    fallbackLng: "en",
    interpolation: {
        escapeValue: false,
    },
});

export default i18n;
