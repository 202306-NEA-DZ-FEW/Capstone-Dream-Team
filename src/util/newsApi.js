export async function fetchData(apiRoute) {
    const url = `https://newsapi.org/v2/${apiRoute}&apiKey=${process.env.NEXT_PUBLIC_NEWS_API}`;
    const response = await fetch(url);
    const data = await response.json();
    return data;
}
