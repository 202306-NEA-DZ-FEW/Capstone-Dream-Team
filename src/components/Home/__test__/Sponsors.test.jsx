import renderer from "react-test-renderer";

import { getAuth } from "@/util/firebase";

import Sponsors from "../Sponsors";

// Mock the Firebase services
jest.mock("@/util/firebase", () => ({
    getAuth: jest.fn(),
}));
jest.mock("next/router", () => ({
    useRouter: jest.fn().mockReturnValue({
        push: jest.fn(),
    }),
}));

it("rend correctement", () => {
    const tree = renderer.create(<Sponsors />).toJSON();
    expect(tree).toMatchSnapshot();
});
