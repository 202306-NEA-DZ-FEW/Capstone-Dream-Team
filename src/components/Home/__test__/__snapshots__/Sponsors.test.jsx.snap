// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rend correctement 1`] = `
<div
  class="max-w-screen-xl mx-auto flex h-full  my-20 py-20 items-center justify-center  bg-[#BCE4EB]"
>
  <div
    class="w-full rounded-lg bg-white px-8 py-4 shadow-md m-4"
  >
    <div
      class="px-1 py-4"
    >
      <h3
        className=" mb-6 font-sans text-2xl text-center font-bold leading-none tracking-tight text-[#192655] sm:text-3xl md:mx-auto"
      >
        home.ourSponsors
      </h3>
    </div>
    <ul
      class="grid grid-cols-4 gap-2 px-1"
    />
  </div>
</div>
`;
